core/src/main/scala/aima/core/search/adversarial/Game.scala

```diff
@@ -1,14 +1,27 @@
 package aima.core.search.adversarial
 
-final case class UtilityValue(value: Double) extends AnyVal
+final case class UtilityValue(value: Double) extends AnyVal {
+  def <(that: UtilityValue)  = if (this.value < that.value) true else false
+  def >(that: UtilityValue)  = if (this.value > that.value) true else false
+  def ==(that: UtilityValue) = if (this.value == that.value) true else false
+
+}
 
 /**
   * @author Aditya Lahiri
+  * @author Shawn Garner
   */
-trait Game[Player, State, Action] {
-  def initialState: State
-  def getPlayer(state: State): Player
-  def getActions(state: State): List[Action]
-  def result(state: State, action: Action): State
-  def getUtility(state: State, player: Player): UtilityValue
+trait Game[P <: Player, S <: State, A <: Action] {
+  def initialState: S
+  def getPlayer(state: S): P
+  def getActions(state: S): List[A]
+  def result(state: S, action: A): S
+  def isTerminalState(state: S): Boolean
+  def getUtility(state: S): UtilityValue
 }
+
+trait Player {}
```

Comment:
I don't really like having traits with no defined behavior.
I'd just as soon keep these in the definition of the given game so they can be sealed and compile time checked to be total matched.